### PR TITLE
Add CI Terraform Plan for EKS

### PR DESCRIPTION
**Pull request by fredericfran-gds**
Since we switch to EKS from ECS, this PR adds the terraform plans for
the EKS modules and removes the ECS ones.

**Testing**

Tested by setting the CI pipeline to the one defined here and
checking that the 2 terraform plans are done and reported to
GitHub

new CI pipeline view

new GitHub checks on PR page

**Notes**
We use plan with `-lock=false` and `-refresh=false` as done
previously in ECS with justifications in #297 and #306 respectively.

Ref:
1. [EKS cluster](#372)
2. [EKS cluster addons](#377)
3. [Concourse EKS pipeline](#378)
4. [add CI user to k8s](#391)

### CODE REVIEW

File: concourse/tasks/terraform-plan.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  | jq .Credentials)"
+export AWS_ACCESS_KEY_ID="$(echo $creds | jq -r .AccessKeyId)"
+export AWS_SECRET_ACCESS_KEY="$(echo $creds | jq -r .SecretAccessKey)"
+export AWS_SESSION_TOKEN="$(echo $creds | jq -r .SessionToken)"
+
+cd "${DEPLOYMENT_PATH}"
+
+terraform init -input=false -backend-config "${ENVIRONMENT}.backend"
+
+terraform plan -input=false \
```
Comment (sengi) [inline concourse/tasks/terraform-plan.sh:24]: Please don't copy-paste this file. All you need to do make the script reusable is to change `plan` to `$1` and pass `plan`/`apply` as a positional argument from Concourse (and rename the file to `terraform.sh`).

Comment (fredericfran-gds) [reply concourse/tasks/terraform-plan.sh:24]: yeah, I know, testing a bit. I don't know if we are interested in that feature too.

Comment (sengi) [reply concourse/tasks/terraform-plan.sh:24]: Cool, no worries then. I might just go ahead and refactor the script anyway - I kinda made it unnecessarily specific before.

Comment (fredericfran-gds) [reply concourse/tasks/terraform-plan.sh:24]: Thanks. I will put it into review once I'm ready, still investigating some k8s auth.

**Comment by fredericfran-gds** [comment]
This work has a dependency on trello card [iam-eks-configure-fulladmin-and-readonly-access-levels](https://trello.com/c/ONhtNSTg/406-iam-eks-configure-fulladmin-and-readonly-access-levels) so we will wait for next sprint.

**Review by sengi — COMMENTED** [review]
If you rebase, there's a refactored version of the Terraform script that you can use.

**Comment by fredericfran-gds** [comment]
> If you rebase, there's a refactored version of the Terraform script that you can use.

I was using that but forgot to remove the old script, will do now.

**Review by sengi — APPROVED** [review]
Cool. Looks good, cheers for getting this running again.

**Pull request merged**